### PR TITLE
Update Target SDK

### PR DESCRIPTION
**Pull request by rajramsaroop**
Updates the app to target sdk level 34, to meet play store requirements.

Fixed a few build issues so it can now run on more recent versions of android studio.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -23,11 +22,23 @@ android {
...
+        viewBinding = true
+    }
+
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+
+    kotlinOptions {
+        jvmTarget = "1.8"
```
Comment (phatblat) [inline app/build.gradle:35]: Is this Java 8 compatibility just to stay the same as the existing release?

**Review by phatblat — APPROVED** [review]
Nice updates!

File: app/build.gradle (hunk 1 above)
Comment (rajramsaroop) [reply app/build.gradle:35]: Yep. I just did the minimum required to get the app to build. I don't see any issue with updating to something newer

Comment (rajramsaroop) [reply app/build.gradle:35]: This also matches what we use in DittoAndroidTools. If we migrate the wifi checker to the tools, we can also update it there.

Comment (phatblat) [reply app/build.gradle:35]: This should be fine.

We're compatible with Java 11 in the SDK and I've been wondering what the impact would be if we bump this to Java 17.
https://github.com/getditto/ditto/blob/3bfc0005b44578bfbf704663f75a3cf501edfd1f/android/ditto/build.gradle.kts#L31-L37

**Pull request merged**